src/erc20.ts

```diff
@@ -1,77 +1,61 @@
 import { ADDRESS_BURN } from "./constants";
 
 export function handleTransfer(event: TransferEvent): void {
-    let contract = fetchERC20(event.address);
-    if (event.params.from == constants.ADDRESS_ZERO) {
-        let totalSupply = fetchERC20Balance(contract, null);
-        totalSupply.valueExact = totalSupply.valueExact.plus(
-            event.params.value
-        );
-        totalSupply.value = decimals.toDecimals(
-            totalSupply.valueExact,
-            contract.decimals
-        );
-        totalSupply.save();
-    } else {
-        let from = fetchAccount(event.params.from);
-        let balance = fetchERC20Balance(contract, from);
-        balance.valueExact = balance.valueExact.minus(event.params.value);
-        balance.value = decimals.toDecimals(
-            balance.valueExact,
-            contract.decimals
-        );
-        balance.save();
-    }
+  let contract = fetchERC20(event.address);
+  if (event.params.from == constants.ADDRESS_ZERO) {
+    let totalSupply = fetchERC20Balance(contract, null);
+    totalSupply.valueExact = totalSupply.valueExact.plus(event.params.value);
+    totalSupply.value = decimals.toDecimals(
+      totalSupply.valueExact,
+      contract.decimals
+    );
+    totalSupply.save();
+  } else {
+    let from = fetchAccount(event.params.from);
+    let balance = fetchERC20Balance(contract, from);
+    balance.valueExact = balance.valueExact.minus(event.params.value);
 
-    if (
-        event.params.to == constants.ADDRESS_ZERO ||
-        event.params.to == ADDRESS_BURN
-    ) {
-        let burnAccount = fetchAccount(ADDRESS_BURN);
-        let burnedBalance = fetchERC20Balance(contract, burnAccount);
+    balance.value = decimals.toDecimals(balance.valueExact, contract.decimals);
 
-        burnedBalance.valueExact = burnedBalance.valueExact.plus(
-            event.params.value
-        );
-        burnedBalance.value = decimals.toDecimals(
-            burnedBalance.valueExact,
-            contract.decimals
-        );
-        burnedBalance.save();
+    balance.save();
+  }
 
-        // remove from total supply
-        let totalSupply = fetchERC20Balance(contract, null);
-        totalSupply.valueExact = totalSupply.valueExact.minus(
-            event.params.value
-        );
-        totalSupply.value = decimals.toDecimals(
-            totalSupply.valueExact,
-            contract.decimals
-        );
-        totalSupply.save();
-    } else {
-        let to = fetchAccount(event.params.to);
-        let balance = fetchERC20Balance(contract, to);
-        balance.valueExact = balance.valueExact.plus(event.params.value);
-        balance.value = decimals.toDecimals(
-            balance.valueExact,
-            contract.decimals
-        );
-        balance.save();
-    }
+  if (
+    event.params.to == constants.ADDRESS_ZERO ||
+    event.params.to == ADDRESS_BURN
+  ) {
+    let burnAccount = fetchAccount(ADDRESS_BURN);
```

Comment:
@cinnabarhorse I think, this line would be problem.
In both cases (zero address and burn address), burned balance is calculated as burn balance. This may cause the issue.
What do you think?

Reply:
Yep that's probably it. That code was already there (not part of my original PR) so you'll need to verify what it's doing before you modify it @orionstardust

Reply:
@cinnabarhorse 
Yes, I checked source code again.
Also checked alchemica token balances of burn tokens for Alchemica tokens. They were correct. (And alchemica token balances of zero address were zero.)
I think, we can deploy test version of fix, and then try.